src/main/java/org/broadinstitute/hellbender/tools/exome/GetBayesianHetCoverage.java

```
@@ -226,73 +296,28 @@ protected Object doWork() {
-                tumorHetPulldown.write(tumorHetOutputFile);
-
+                runTumorOnly();
                 break;
 
-            case NORMAL_TUMOR:
-
-                normalHetPulldownCalculator = new BayesianHetPulldownCalculator(REFERENCE_ARGUMENTS.getReferenceFile(),
-                        IntervalList.fromFile(snpFile), minimumMappingQuality, minimumBaseQuality, readDepthThreshold,
-                        VALIDATION_STRINGENCY, errorProbabilityAdjustmentFactor);
-
-                logger.info("Calculating the Het pulldown from the normal BAM file using the BALANCED prior...");
-                normalHetPulldown = normalHetPulldownCalculator.getHetPulldown(normalBamFile, hetCallingStringency);
-
-                logger.info("Writing Het pulldown from normal reads to " + normalHetOutputFile.toString());
-                normalHetPulldown.write(normalHetOutputFile);
-
-                tumorHetPulldownCalculator = new BayesianHetPulldownCalculator(REFERENCE_ARGUMENTS.getReferenceFile(),
-                        normalHetPulldown.getIntervals(), minimumMappingQuality, minimumBaseQuality, readDepthThreshold,
-                        VALIDATION_STRINGENCY, errorProbabilityAdjustmentFactor);
-
-                logger.info("Calculating the Het pulldown from the tumor BAM file on Hets detected in the normal BAM file...");
-                tumorHetPulldown = tumorHetPulldownCalculator.getTumorHetPulldownFromNormalPulldown(tumorBamFile,
-                        normalHetPulldown);
-
-                logger.info("Writing Het pulldown from tumor reads to " + tumorHetOutputFile.toString());
-                tumorHetPulldown.write(tumorHetOutputFile);
-
+            case MATCHED_NORMAL_TUMOR:
+                runMatchedNormalTumor();
                 break;
 
             case NONE:
-                break;
+                throw new GATKException.ShouldNeverReachHereException("The job type is not properly determined from" +
```

Can you actually throw a user exception telling the user what was the problem?.